### ISSUE TITLE
Request:  Paste/Fetch - allow 1 upload job per URL instead of 1 job for all URLs

### ISSUE THREAD

**Issue by mblue9** [galaxyproject/galaxy#6473]
Hello,

I was importing some files from SRA with Paste/Fetch, by pasting the URLs into the text box, and noticed that if I pasted all links (12 files) into the URL box (see screenshot below) it seems to run as 1 big job that took > 40 mins:

<img width="786" alt="screen shot 2018-07-08 at 12 28 28 pm" src="https://user-images.githubusercontent.com/5591821/42416153-8c1b2fe4-82aa-11e8-97cc-8a4f99f0c83c.png">

And if I pasted one URL per text box like below I could get the files to upload as separate jobs that then took less than 10 mins to the load the 12 files.

<img width="888" alt="screen shot 2018-07-07 at 7 49 44 pm" src="https://user-images.githubusercontent.com/5591821/42416155-a4323aa0-82aa-11e8-9556-0a6c7f9193bd.png">

I like the faster method but it's a little bit of a pain to have to copy each link into a separate text box in the Paste/Fetch so wondering if there could be an option for pasting all URLs into 1 text box (like in the first screenshot) and then having 1 upload job per URL instead of 1 job for all?

(with the 1 upload job per URL, as in the second screenshot above, I think it's also nice to see the datasets turning green quickly in the history as each file uploads, rather than having all datasets spinning yellow in the history for > 40 mins & not knowing if the upload is working or not, which is what happens with the method in the first screenshot)

Some info:

First screenshot leads to upload of all URLs together:
python '/usr/local/galaxy/galaxy-dist/tools/data_source/upload.py' /usr/local/galaxy/galaxy-dist /data/dnb01/galaxy_db/job_working_directory/004/057/4057529/registry.xml /data/dnb01/galaxy_db/job_working_directory/004/057/4057529/upload_params.json 7050403:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050403_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050403.dat 7050404:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050404_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050404.dat 7050405:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050405_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050405.dat 7050406:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050406_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050406.dat 7050407:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050407_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050407.dat 7050408:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050408_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050408.dat 7050409:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050409_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050409.dat 7050410:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050410_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050410.dat 7050411:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050411_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050411.dat 7050412:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050412_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050412.dat 7050413:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050413_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050413.dat 7050414:/data/dnb01/galaxy_db/job_working_directory/004/057/4057529/dataset_7050414_files:/data/dnb01/galaxy_db/files/007/050/dataset_7050414.dat

Files I was using:
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/004/SRR1552444/SRR1552444.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/005/SRR1552445/SRR1552445.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/006/SRR1552446/SRR1552446.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/007/SRR1552447/SRR1552447.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/008/SRR1552448/SRR1552448.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/009/SRR1552449/SRR1552449.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/000/SRR1552450/SRR1552450.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/001/SRR1552451/SRR1552451.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/002/SRR1552452/SRR1552452.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/003/SRR1552453/SRR1552453.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/004/SRR1552454/SRR1552454.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR155/005/SRR1552455/SRR1552455.fastq.gz

**Comment by martenson** [comment]
'All files in one request' is a fairly recent change that solved a bunch of issues Galaxy had with uploads. It could be improved further to address your points though. The actual bottleneck in the case you described could very well be the bandwidth the remote server (SRA here) gives per connection. However the probable culprit is that the metadata detection is being run after all the files are downloaded so they 'wait' for each other before turning green.

To sum it up imho the best scenario would be to have API that consumes list of URLs but is capable of splitting the fetching into multiple jobs. I can imagine having the splitting as an option if people deem it unreasonable default.

**Comment by mblue9** [comment]
@martenson sounds good, thanks for considering!
